### PR TITLE
feat: add ai chat block

### PR DESCRIPTION
**Pull request by donteatfriedrice**

### CODE REVIEW

File: packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      : html`<user-info
+          .userName=${'AFFiNE AI'}
+          .avatarIcon=${AffineAIIcon}
+        ></user-info>`;
+
+    return userInfoTemplate;
+  }
+
+  override renderBlock() {
+    const messages = this._deserializeChatMessages(this.model.messages).slice(
```
Comment (Saul-Mirone) [inline packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts:70]: Recommend to use `this.model.messages$.value` to let lit subscribe the prop changes.

**Comment by doouding** [comment]
LGTM

File: packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import './components/text-renderer.js';
+import './components/user-info.js';
+import { styles } from './styles.js';
+import { type ChatMessage, ChatMessagesSchema } from './types.js';
+
+@customElement('affine-ai-chat')
+@Peekable()
+export class AIChatBlockComponent extends BlockElement<AIChatBlockModel> {
+  // Deserialize messages from JSON string and verify the type using zod
+  private _deserializeChatMessages = (messages: string) => {
```
Comment (Saul-Mirone) [inline packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts:20]:
```suggestion
  private _deserializeChatMessages = (messages: string) => {
  private _deserializedChatMessaged = computed(() => {
```

File: packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    const result = ChatMessagesSchema.safeParse(JSON.parse(messages));
```
Comment (Saul-Mirone) [inline packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts:21]:
```suggestion
    const result = ChatMessagesSchema.safeParse(JSON.parse(this.model.messages$.value));
```

File: packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    if (result.success) {
+      return result.data;
+    } else {
+      return [];
+    }
+  };
```
Comment (Saul-Mirone) [inline packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts:27]:
```suggestion
  });
```

File: packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      : html`<user-info
+          .userName=${'AFFiNE AI'}
+          .avatarIcon=${AffineAIIcon}
+        ></user-info>`;
+
+    return userInfoTemplate;
+  }
+
+  override renderBlock() {
+    const messages = this._deserializeChatMessages(
```
Comment (Saul-Mirone) [inline packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts:70]:
```suggestion
    const messages = this._deserializedChatMessaged.value.slice(-2);
```

File: packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      this.model.messages$.value
+    ).slice(-2);
```
Comment (Saul-Mirone) [inline packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts:72]:
```suggestion
    const messages = this._deserializedChatMessaged.value.slice(-2);
```

File: packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts (hunk 4 above)
Comment (Saul-Mirone) [inline packages/presets/src/blocks/ai-chat-block/ai-chat-block.ts:27]:
```suggestion
  private _deserializeChatMessages = computed(() => {
    const messages = this.model.messages$.value
    try {
      const result = ChatMessagesSchema.safeParse(JSON.parse(messages));
      if (result.success) {
        return result.data;
      } else {
        return [];
      }
    } catch {
      return [];
    }
  };
```

**Review by Saul-Mirone — CHANGES_REQUESTED** [review]

**Review by Saul-Mirone — APPROVED** [review]

**Pull request merged**